libraries/Microsoft.Bot.Builder.Dialogs/DialogManager.cs

```diff
@@ -144,8 +144,12 @@ private static async Task<StoredBotState> LoadBotState(IStorage storage, BotStat
             };
         }
 
-        private static async Task SaveBotState(IStorage storage, StoredBotState newState, BotStateStorageKeys keys)
+        private static async Task SaveBotState(IStorage storage, StoredBotState newState, BotStateStorageKeys keys, DialogContext dialogContext)
         {
+            // Save dialog and turn memory to the conversation state for inspection.
+            newState.ConversationState["dialog"] = dialogContext.DialogState;
+            newState.ConversationState["turn"] = dialogContext.State.Turn;
```

Comment:
no this is definitly not right.  Putting turn into the conversationstate means it will store it to storage, which is the opposite of what turn state is supposed to be.

Reply:
I have updated based on your recommendation, but I'am not sure whether this is a good solution. agreed with @vishwacsena, we should not show them all. Any suggestions on that?